### PR TITLE
ex6 addition: this example performs the analysis for different solver…

### PR DESCRIPTION
**Pull request by rprat-pro**
I push a first version of this example. It still lacks the doxygene doc, the tests with PETSC and the feedback on Topaze in a markdown file.

solver tested:

* HyprePCG, HypreFGMRES, HypreGMRES, MUMPSSolver, CGSolver, BiCGSTABSolver, MINRESSolver

preconditioner tested:

* ANY, HypreBoomerAMG, HypreILU, HypreEuclid, HypreParaSails, HypreDiagScale

### CODE REVIEW

**Review by latug0 — CHANGES_REQUESTED** [review]

File: ex6/timer.hpp [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+			for(auto& it:daughters)
+				recursive_sorted_call(func, mySort, it,arg...);
+		}
+
+
+	};
+};
+
+
+#define START_TIMER(name) profiling::timers::profilingTimer*& current = profiling::timers::get_timer<CURRENT>(); \
```
[the remark this answers is not in the document]
Comment (rprat-pro) [reply ex6/timer.hpp:200]: Sure I can.

File: ex6/timer.hpp [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+#include<iostream>
+#include<chrono>
+#include<vector>
+#include<omp.h>
+#include <fstream>
+#include <algorithm>
+#include<cassert>
+
+// variables 
+#ifdef __MPI
```
[the remark this answers is not in the document]
Comment (rprat-pro) [reply ex6/timer.hpp:12]: Currently, I would like to re-use timer.hpp and timer.cpp for other codes. As these timers are not included in mfem-mgis, I preferred to manage it with the cmake file for now.

File: ex6/Feedback-topaze.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
+
```
[the remark this answers is not in the document]
Comment (rprat-pro) [reply ex6/Feedback-topaze.md:1]: Yes, this file is still in construction (I'm still fighting my bug with 64 mpi processes).

File: ex6/perf.cxx [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+		solver_name::CGSolver,
+		solver_name::BiCGSTABSolver,
+		solver_name::MINRESSolver
+	};
+
+	std::vector<precond_name> precondTraversal = 
+	{
+		precond_name::ANY,
+		precond_name::HypreBoomerAMG,
+		precond_name::HypreILU,
```
[the remark this answers is not in the document]
Comment (rprat-pro) [reply ex6/perf.cxx:255]: I'm gonna add it

**Pull request merged**